src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java

```diff
@@ -326,6 +326,26 @@ public SerializationPair<Object> getValueSerializationPair() {
 		return this.valueSerializationPair;
 	}
 
+	/**
+	 * Returns a computed {@link Duration TTL expiration timeout} based on cache entry key/value
+	 * if a {@link TtlFunction} was confiugred using {@link #entryTtl(TtlFunction)}.
+	 * <p>
+	 * Otherwise, returns the user-provided, fixed {@link Duration} if {@link #entryTtl(Duration)}
+	 * was called during cache configuration.
+	 *
+	 * @return the configured {@link Duration TTL expiration}.
+	 * @deprecated use {@link #getTtlFunction()} instead.
+	 */
+	@Deprecated
+	@SuppressWarnings("all")
+	public Duration getTtl() {
+
+		TtlFunction ttlFunction = getTtlFunction();
+
+		return ttlFunction instanceof FixedDurationTtlFunction it ? it.duration()
+			: ttlFunction.getTimeToLive(null, null);
```

Comment:
Why a `null` key?

Because this is what the old, mistakingly removed, now deprecated (once again) `getTtl()` method [did](https://github.com/spring-projects/spring-data-redis/commit/770fbac67094ee0d570a17ba73243cfad6892df5#diff-9b091dbe458d10f65774fb0a47e282e73b8535d4d9ae178342041badf6fb3e46L331) with the introduction of dynamic TTL cache configuration using the new `RedisCacheWriter.TtlFunction` interface.

Reply:
Why case to `FixedDurationTtlFunction`?

Because it is still possible to configure a "fixed" `Duration` TTL cache entry expiration policy using the overloaded `entryTtl(:Duration)` ([source](https://github.com/spring-projects/spring-data-redis/blob/main/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L200-L211)) builder method.

> NOTE: `entryTtl(:Duration)` is an alternative to `entryTtl(:TtlFunction)` ([here](https://github.com/spring-projects/spring-data-redis/blob/main/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L213-L227)), hence the overload.

From this [line](https://github.com/spring-projects/spring-data-redis/blob/main/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L210), it will then  create a (non-dynamically computed) `FixedDurationTtlFunction` instance ([here](https://github.com/spring-projects/spring-data-redis/blob/main/src/main/java/org/springframework/data/redis/cache/RedisCacheWriter.java#L185-L197)).

Reply:
`FixedDurationTtlFunction#getTimeToLive` would return the duration nevertheless, so really I do not see the need for a cast here.

Reply:
Outside of a potential JVM runtime optimization here, it is true, there probably is not a need to cast to a `FixedDurationTtlFunction` and call the `duration()` method directly.

Feel free to change it if you like.